js/sites.js

```diff
@@ -203,11 +203,42 @@
       let enhanced = this.link;
       let url = this.url;
 
+      var getRegularThumbnail = async(function*() {
+        var sw = (yield navigator.serviceWorker.ready).active;
+        var host = new URL(url).host;
+        var thumbURL = new URL(`/pagethumbs/${host}`, window.location);
+
+        var id = Math.random();
```

Comment:
future developers will be glad to have a description of why an id is generated here.

cross-page communication is hard to describe and we should be pro-active about it